tests/unit/api/bucketPutEncryption.js

```diff
@@ -0,0 +1,310 @@
+    afterEach(() => cleanup());
+
+    describe('test invalid sse configs', () => {
+        it('should reject an empty config', done => {
+            bucketPutEncryption(authInfo, templateRequest({ post: '' }), log, err => {
+                assert.strictEqual(err.MalformedXML, true);
+                done();
+            });
+        });
+
+        it('should reject a config with no Rule', done => {
+            bucketPutEncryption(authInfo, templateRequest(
+            { post: `<?xml version="1.0" encoding="UTF-8"?>
+                <ServerSideEncryptionConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                </ServerSideEncryptionConfiguration>`,
+            }), log, err => {
+                assert.strictEqual(err.MalformedXML, true);
+                done();
+            });
+        });
+
+        it('should reject a config with no ApplyServerSideEncryptionByDefault section', done => {
+            bucketPutEncryption(authInfo, templateRequest(
+            { post: `<?xml version="1.0" encoding="UTF-8"?>
+                <ServerSideEncryptionConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+                <Rule></Rule>
+                </ServerSideEncryptionConfiguration>`,
+            }), log, err => {
+                assert.strictEqual(err.MalformedXML, true);
+                done();
+            });
+        });
+
+        it('should reject a config with no SSEAlgorithm', done => {
+            const post = templateSSEConfig({});
+            bucketPutEncryption(authInfo, templateRequest({ post }), log, err => {
+                assert.strictEqual(err.MalformedXML, true);
+                done();
+            });
+        });
+
+        it('should reject a config with an invalid SSEAlgorithm', done => {
+            const post = templateSSEConfig({ algorithm: 'InvalidAlgo' });
+            bucketPutEncryption(authInfo, templateRequest({ post }), log, err => {
+                assert.strictEqual(err.MalformedXML, true);
+                done();
+            });
+        });
+
+        it('should reject a config with SSEAlgorithm == AES256 and a provided KMSMasterKeyID', done => {
+            const post = templateSSEConfig({ algorithm: 'AES256', keyId: '12345' });
+            bucketPutEncryption(authInfo, templateRequest({ post }), log, err => {
+                assert.strictEqual(err.InvalidArgument, true);
+                done();
+            });
+        });
+    });
+
+    describe('test setting config without a prvious one', () => {
+        it('should apply a config with SSEAlgorithm == AES256', done => {
+            const post = templateSSEConfig({ algorithm: 'AES256' });
+            bucketPutEncryption(authInfo, templateRequest({ post }), log, err => {
+                assert.ifError(err);
+                return getSSEConfig((err, sseInfo) => {
+                    assert.ifError(err);
+                    assert.notStrictEqual(sseInfo, null);
```

Comment:
Those two first checks are not needed since there is a `deepStrictEqual` afterwards, regarding ease of debugging it should be fine since `deepStrictEqual` should show the actual diff between what's expected and what's obtained. Same comment for following tests.